angel_system/eval/visualization.py

```diff
@@ -4,74 +4,161 @@
+                    ys_bottom = [0] #[1.01]
+                    # TODO: Make this something that is added be clicking?
+                    ax.bar(xs_bars2, ys_height,
+                        width=bar_widths2, bottom=ys_bottom, align="edge",
+                        color=custom_range_color, alpha=0.5)
+
+                # ============================
+                # Detections
+                # ============================
+                xs2_bars = [p["time"][0] for p in det_ranges]
+                ys2_det_regions = [p["conf"] for p in det_ranges]
+                bar_widths2 = [(p["time"][1] - p["time"][0]) for p in det_ranges]
+                ax.bar(xs2_bars, ys2_det_regions, width=bar_widths2, align="edge", edgecolor="blue", fill=False, label="Detections")
+
+                ax.legend(loc="upper right")
+                ax.plot
+
+                # ============================
+                # Save
+                # ============================
+                #plt.show()
+                activity_plot_dir = Path(os.path.join(self.output_dir, "activities"))
+                activity_plot_dir.mkdir(parents=True, exist_ok=True)
+                fig.savefig(f"{str(activity_plot_dir)}/{label.replace(' ', '_')}.png")
+            else:
+                log.warning(f"No detections/gt found for \"{label}\"")
+
+    def plot_pr_curve(self, iou_thr=0.0):
+        """
+        Plot the PR curve for each label
+        
+        :param iou_thr: IoU threshold
+        """
+        # ============================
+        # Setup figure
+        # ============================
+        fig, ax = plt.subplots(figsize=(7, 8))
+
+        ax.set_xlim([0.0, 1.0])
+        ax.set_ylim([0.0, 1.05])
+        ax.set_title("Precision vs. Recall")
+
+        colors = plt.cm.rainbow(np.linspace(0, 1, len(self.labels)))
+        
+        # ============================
+        # Add F1 score 
+        # ============================
+        fscores = np.linspace(0.2, 0.8, num=4)
+        for f_score in fscores:
+            x = np.linspace(0.01, 1)
+            y = f_score * x / (2 * x - f_score)
+            (l,) = plt.plot(x[y >= 0], y[y >= 0], color="gray", alpha=0.2)
+            plt.annotate("f1={0:0.1f}".format(f_score), xy=(0.9, y[45] + 0.02))
+
+        # ============================
+        # Get PR plot per class 
+        # ============================
+        precision = dict()
+        recall = dict()
+        thresholds = dict()
+        average_precision = dict()
+        for i, row in self.labels.iterrows():
+            id = row['id']
+            label = row['class']
+
+            truth = [1 if det['iou'] > iou_thr else 0 for det in self.dets[label]]
+            pred = [det['conf'] for det in self.dets[label]]
+
+            precision[i], recall[i], thresholds[i] = precision_recall_curve(truth, pred)
+            average_precision[i] = average_precision_score(truth, pred)
+
+            display = PrecisionRecallDisplay(
```

Comment:
Usage of this class is new to me! Is this to provide a convenient plot formatting with the axis and legend like in the scikit-learn docs page?

In the docs page, it does say that [`from_predictions`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.PrecisionRecallDisplay.html#sklearn.metrics.PrecisionRecallDisplay.from_predictions) is recommended instead of directly creating this class. It might be simpler to call that here instead of calling `precision_recall_curve` and `average_precision_score` manually (it calls those methods for you).